labkey/query.py

```diff
@@ -170,6 +170,7 @@ def delete_rows(
     query_name: str,
     rows: any,
     container_path: str = None,
+    options: dict = None,
```

Comment:
@labkey-nicka I am curious about your thoughts on this.

Rosaline says there are about 8 options in total we want to support, so she added an "options" dict to the various API methods, instead of adding them as additional kwargs. Our JS API seems to have these options right in the QueryRequestOptions definition, which makes me think we should just add them all as kwargs. However, adding 8 additional kwargs makes these API method definitions look really noisy.

Reply:
**tl;dr** It feels a little too untyped/undiscoverable for my liking. I prefer explicit kwargs.

I understand the sentiment and certainly agree a lot of parameters are not supported in this package that are supported by the endpoints, however, I feel like Python's type system is not quite up to the task yet (but maybe there are things I don't know about). I feel like we should either commit to all explicit keyword parameters or all `options` (for optional parameters) with a more refined type than `dict` (here is an interesting [PEP](https://peps.python.org/pep-0692/)). As it stands with this PR some optional things are keyword parameters and some are `options` and it is unclear why some are "in" and some are "out" of `options`. Additionally, the user has no way of exploring the options from the code whereas the keyword parameters are documented inline. If we could get typed options, and IDEs are able to explore them, then I'd be more inclined toward generic `options`.

I'll also note that `QueryRequestOptions` is not really a great type either for describing the exact endpoint supported parameters (for example, `transact` is not supported for update so it doesn't feel like it belongs there, `skipReselectRows` only applies to insert/update -- not delete, etc).

Reply:
Ok, I think with Nick's feedback we should go the explicit kwargs route, and type them all as best we can.